server/api/api/endpoints/nuclio.py

```diff
@@ -455,7 +455,7 @@ def _deploy_function(
         # Enrich runtime
         launcher = server.api.launcher.ServerSideLauncher(auth_info=auth_info)
         launcher.enrich_runtime(runtime=fn, full=True)
-
+        fn.pre_save_validation()
         fn.save(versioned=False)
         fn.pre_deploy_validation()
```

Comment:
Why not just move the `pre_deploy_validation` before the `save`?

Reply:
Because it is validation before saving and not before deploying.
I also thought about renaming `pre_deploy_validation` to `pre_save_validation` but again, it is not good because we call this function before deploying and not before saving.

Reply:
Spoke offline, tag validation can be moved here: https://github.com/mlrun/mlrun/blob/development/server/api/db/sqldb/db.py#L1687